ocaml/xapi/xapi_bond.ml

```diff
@@ -431,6 +431,21 @@ let create ~__context ~network ~members ~mAC ~mode ~properties =
             Db.PIF.set_disallow_unplug ~__context ~self:pif ~value:false)
           members
       end;
+
+      (* If the host has a cluster_host AND the cluster_host's PIF is a member, update the PIF *)
+      begin match Xapi_clustering.find_cluster_host ~__context ~host with
+      | Some cluster_host ->
+        let cluster_pif_ref = Db.Cluster_host.get_PIF ~__context ~self:cluster_host in
+        if List.mem cluster_pif_ref members
+        then begin
+          let master_pif_rec = Db.PIF.get_record ~__context ~self:master in
+          Xapi_clustering.assert_pif_prerequisites (master,master_pif_rec);
+          debug "Setting cluster_host %s PIF to new bond PIF %s" (Ref.string_of cluster_host) (Ref.string_of master);
+          Db.Cluster_host.set_PIF ~__context ~self:cluster_host ~value:master
+        end
+      | None -> ()
+      end;
+
       TaskHelper.set_progress ~__context 1.0;
```

Comment:
You should refine the progress updates since there is a new step introduced now.

Reply:
Is this just for a progress bar? I.e. are the intervals meaningful or can I just put something in between 0.8 and 1.0, since moving the cluster PIF isn't too big a step?

Reply:
Yes, you can make up a number to more or less indicate the amount of process.